auth/src/main/resources/create-rmap-agent.sql

```diff
@@ -1,7 +1,20 @@
-INSERT into Users(name, email, isActive, rmapAgentUri, authKeyUri, doRMapAgentSync, createdDate, lastAccessedDate) values ('RMap Test User', 'testuser@example.com', TRUE, 'rmap:testagenturi', 'http://rmap-hub.org/authids/testauthid', TRUE, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP);
+INSERT into Users(name, email, isActive, rmapAgentUri, authKeyUri, doRMapAgentSync, createdDate, lastAccessedDate)
+SELECT
+  'RMap Test User', 'testuser@example.com', TRUE, 'rmap:testagenturi', 'http://rmap-hub.org/authids/testauthid', TRUE, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+WHERE NOT EXISTS (
+  SELECT * FROM Users WHERE rmapAgentUri = 'rmap:testagenturi'
+);
 
 INSERT into UserIdentityProviders(identityProvider, providerAccountId, providerAccountPublicId, providerAccountDisplayName, providerAccountProfileUrl, userId, createdDate, LASTAUTHENTICATEDDATE)
-values ('http://exampleidprovider.org', '1234567890', 'rmaptestuser', 'RMap Test User', 'https://exampleidprovider.org/rmaptestuser', (select userId from Users where name='RMap Test User'), CURRENT_TIMESTAMP, CURRENT_TIMESTAMP);
+SELECT
+  'http://exampleidprovider.org', '1234567890', 'rmaptestuser', 'RMap Test User', 'https://exampleidprovider.org/rmaptestuser', (select userId from Users where name='RMap Test User'), CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+WHERE NOT EXISTS (
+  SELECT * FROM UserIdentityProviders WHERE providerAccountId = '1234567890'
+);
 
 INSERT into ApiKeys(accessKey, secret, keyUri, label, note, keyStatus, includeInEvent, userId, createdDate, lastModifiedDate)
-values ('uah2CKDaBsEw3cEQ', 'NSbdzctrP46ZvhTi', 'rmap:testkeyuri', 'RMap test key', 'Test key for RMap', 'ACTIVE', FALSE, (select userId from Users where name='RMap Test User'), CURRENT_TIMESTAMP, CURRENT_TIMESTAMP);
\ No newline at end of file
+SELECT
+  'uah2CKDaBsEw3cEQ', 'NSbdzctrP46ZvhTi', 'rmap:testkeyuri', 'RMap test key', 'Test key for RMap', 'ACTIVE', FALSE, (select userId from Users where name='RMap Test User'), CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+WHERE NOT EXISTS (
+  SELECT * FROM ApiKeys WHERE keyUri = 'rmap:testkeyuri'
+);
```

Comment:
I can look at moving this, but it was originally in src/main, so I'm not super keen ATM to move it out.  Right now the integration tests assume this is on the production classpath.

Reply:
@karenhanson I'd suggest that at this point an issue be created for this.  For a couple of reasons:
#. currently integration tests require this to be on the production class path
#. this file is repeated in multiple places

Because there's a couple of messy things wrapped up in this, I'd prefer to merge this PR and create an issue that:
#. Reduces the copies of `create-rmap-agent.sql` from 3 to 1
#. Insure that integration tests can access this SQL as a test resource, and take it off the production classpath

Reply:
if it won't do harm, then that's ok... just wouldn't want a new build to automatically have this

Reply:
Not sure I understand, but all builds up to this point have had a version of this file, so it's not doing any more harm than it was before :)

Reply:
Tracked in #212

Reply:
i think our messages crossed in cyberspace... that sounds fine. didn't realize this had been there the whole time, but yeah an issue for it sounds right.